WordPress/Classes/ViewRelated/Views/WPRichText/WPRichContentView.swift

```diff
@@ -155,6 +166,27 @@ class WPRichContentView: UITextView {
         }
     }
 
+    private func setupTouchDetection() {
+          for gesture in gestureRecognizers ?? [] {
+              gesture.require(toFail: linkTapGestureRecognizer)
+          }
+          addGestureRecognizer(linkTapGestureRecognizer)
+      }
+
+      @objc func tapRecognized(_ recognizer: UIGestureRecognizer) {
```

Comment:
I haven't found an example where this goes wrong, but just in case, we should be implementing `gestureRecognizer(_:shouldReceive:)` in our delegate to make sure our recognizer only responds to taps on links.

I've been playing with it and this seems to work:

```swift
    func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldReceive touch: UITouch) -> Bool {
        guard gestureRecognizer == linkTapGestureRecognizer else {
            return true
        }
        let point = touch.location(in: self)
        let characterIndex = self.layoutManager.characterIndex(for: point, in: self.textContainer, fractionOfDistanceBetweenInsertionPoints: nil)

        let linkAttribute = self.attributedText?.attribute(.link, at: characterIndex, effectiveRange: nil)
        return linkAttribute is URL
    }
```

Reply:
And what about taps on images? I guess that's related to our comments above?

Reply:
That code doesn't handle the images because I thought we didn't need to, but it's possible as well, just a bit more complicated. The logic is similar to the `tapRecognized`, except it only has to return if the touch should be handled.